lms/djangoapps/teams/plugins.py

```diff
@@ -31,3 +38,38 @@ def is_enabled(cls, course, user=None):
             return False
 
         return is_feature_enabled(course)
+
+
+class TeamsCourseApp(CourseApp):
+    """
+    Course app for teams.
+    """
+
+    app_id = "teams"
+    name = ugettext_noop("Teams")
+    description = ugettext_noop("Leverage teams to allow learners to connect by topic of interest.")
+
+    @classmethod
+    def is_available(cls, course_key: CourseKey) -> bool:
+        """
+        The teams app is currently available globally based on a feature setting.
+        """
+        return settings.FEATURES.get("ENABLE_TEAMS", False)
+
+    @classmethod
+    def is_enabled(cls, course_key: CourseKey) -> bool:
+        return CourseOverview.get_from_id(course_key).teams_enabled
+
+    @classmethod
+    def set_enabled(cls, course_key: CourseKey, enabled: bool, user: 'User') -> bool:
+        raise ValueError("Teams cannot be enabled/disabled via this API.")
```

Comment:
I think we need to keep the implementation consistent. The wikiApp also cannot be enabled/disabled but it does not raise valueError.

Reply:
Yup, that is a mistake, I will fix that.

Reply:
No worries 👍